packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -123,11 +124,52 @@ class CupertinoCheckbox extends StatefulWidget {
 
   /// The color to use when this checkbox is checked.
   ///
+  /// If [fillColor] returns a non-null color in the [WidgetState.selected]
+  /// state, it will be used instead of this color.
+  ///
   /// Defaults to [CupertinoColors.activeBlue].
   final Color? activeColor;
 
+  /// {@template flutter.cupertino.CupertinoCheckbox.fillColor}
+  /// The color used to fill this checkbox.
+  ///
+  /// Resolves in the following states:
+  ///  * [WidgetState.selected].
+  ///  * [WidgetState.hovered].
+  ///  * [WidgetState.focused].
+  ///  * [WidgetState.disabled].
+  ///
+  /// {@tool snippet}
+  /// This example resolves the [fillColor] based on the current [WidgetState]
+  /// of the [CupertinoCheckbox], providing a different [Color] when it is
+  /// [WidgetState.disabled].
+  ///
+  /// ```dart
+  /// CupertinoCheckbox(
+  ///   value: true,
+  ///   onChanged: (_){},
+  ///   fillColor: WidgetStateProperty.resolveWith<Color>((Set<WidgetState> states) {
+  ///     if (states.contains(WidgetState.disabled)) {
+  ///       return Colors.orange.withOpacity(.32);
+  ///     }
+  ///     return Colors.orange;
+  ///   })
+  /// )
+  /// ```
+  /// {@end-tool}
+  /// {@endtemplate}
+  ///
+  /// If null, then the value of [activeColor] is used in the selected
+  /// state. If that is also null, then [CupertinoColors.inactiveGray] is used in
+  /// the unselected state, and [CupertinoColors.activeBlue] is used in the
+  /// selected state.
```

Comment:
I suggest the following way:
```suggestion
  /// If [fillColor] resolves to null for the requested state, then the fill color falls back to
  /// [activeColor] if the state includes [WidgetState.selected], or [inactiveColor] otherwise.
```

The differences are:
* It made it clearer on _what_ is null.
* It removes the duplicate mention of the default values of [activeColor] and [inactiveColor].
* It is clearer on what is "selected/unselected state".